sgt-back-end/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+    from "grades"`;
+    db.query(sql)
+    .then((result => {
+      res.status(200).json(result.rows);
+    }))
+    .catch(err => {
+      res.status(500).json({
+        error: 'Something went wrong'
+      })
+    })
+})
+
+app.use(express.json())
+
+app.post('/api/grades', (req, res) => {
+  if(req.body.name === undefined || req.body.course === undefined || req.body.score === undefined
+    || req.body.score < 1 || req.body.score > 100 || !Number.isInteger(parseInt(req.body.score))) {
+    res.status(400).json({
+      error: 'Something is wrong with your request parameters'
+    })
+    return;
+  }
+  const text = `
+  INSERT INTO grades (name, course, score)
+    VALUES($1, $2, $3)
+    RETURNING *`
+  const values = [req.body.name, req.body.course, req.body.score]
+  db.query(text, values)
+  .then((result) => {
+    res.status(201).json(result.rows)
+  })
+  .catch((err) => {
+    res.status(500).json({
+      error: 'Something went wrong'
+    })
+  })
+})
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const gradeId = parseInt(req.params.gradeId)
+  if (!Number.isInteger(gradeId) || gradeId <= 0 || req.body.name === undefined || req.body.course === undefined
+    || req.body.score === undefined || req.body.score < 1 || req.body.score > 100 || !Number.isInteger(parseInt(req.body.score))) {
+      res.status(400).json({
+        error: 'Something is wrong with your request'
+      })
+      return;
+  }
```

Comment:
Same as above: the request should fail if either `gradeId` or `score` isn't an integer, and `parseInt` drops the fractional part before its verified.